controllers/secretsync/secret_sync.go

```diff
@@ -52,7 +52,7 @@ type SecretReconciler struct {
 }
 
 // WARNING: In production, this should be namespaced to the actual managed cluster namespace.
-//+kubebuilder:rbac:groups=core,resources=secrets,verbs=create
+//+kubebuilder:rbac:groups=core,resources=secrets,verbs=create;list;watch
```

Comment:
I added the list and watch verbs because the controllers are caching secrets on the hub and managed clusters. I believe controller-runtime caches uses a list and a watch under the hood so not having these permissions initially I got a lot of `Failed to watch secret` errors when I ran the E2Es. However, I don't think our E2Es catches these errors (probably intended) but I still wanted to make sure the controller has all the correct permissions it needs.

Reply:
It's definitely a good thing to fix, but my point is that the secret it's probably trying to list/watch is only the `policy-encryption-key`, so I think you should be able to add this and still have it work:
```suggestion
//+kubebuilder:rbac:groups=core,resources=secrets,verbs=create
//+kubebuilder:rbac:groups=core,resources=secrets,resourceNames=policy-encryption-key,verbs=list;watch
```